odoo/addons/base/tests/test_ir_sequence.py

```diff
@@ -205,3 +205,49 @@ def test_00(self):
         # Read the value of the current sequence
         n = seq.next_by_id()
         self.assertEqual(n, "0001", 'The actual sequence value must be 1. reading : %s' % n)
+
+
+class TestIrSequenceCacheClearing(common.TransactionCase):
+    def setUp(self):
+        super().setUp()
+        self.company_id = 1
+
+        self.IrSequence = self.env['ir.sequence']
+
+        # Testing invalidates caches randomly, so clear them before doing anything. In particular this means
+        # that calling reset_changes again does nothing unless the cache is invalidated by something we do.
+        self.IrSequence.pool.reset_changes()
+
+        self.seq = self.IrSequence.create({
+            'company_id': self.company_id,
+            'name': 'Test Sequence',
+            'code': 'test.sequence',
+        })
+
+    def test_create_ir_sequence_clears_cache(self):
+        """Check that creating a sequence then rolling back the transaction does not result in stale
+        cache entries, as long as the registry is reset."""
+        _ = self.IrSequence.seq_by_code('test.sequence', self.company_id)
+        self.cr.rollback()
```

Comment:
Is it worth structuring this as an exception inside a savepoint (instead of an explicit rollback)?  That would leave the cursor in the state that TransactionCase would expect to find it.

Reply:
Agreed. See latest (unsquashed) commit.